src/SnowflakeTransformation.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\SnowflakeTransformation;
+
+use Keboola\Component\UserException;
+use Keboola\SnowflakeDbAdapter\Connection;
+use Keboola\SnowflakeDbAdapter\Exception\CannotAccessObjectException;
+use Keboola\SnowflakeDbAdapter\QueryBuilder;
+use Keboola\SnowflakeTransformation\Exception\DeadConnectionException;
+use Psr\Log\LoggerInterface;
+use Retry\BackOff\ExponentialBackOffPolicy;
+use Retry\Policy\SimpleRetryPolicy;
+use Retry\RetryProxy;
+
+class SnowflakeTransformation
+{
+    /** @var Connection $connection */
+    private $connection;
+
+    /** @var LoggerInterface $logger */
+    private $logger;
+
+    /** @var array $databaseConfig */
+    private $databaseConfig;
+
+    public function __construct(Config $config, LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+        $this->databaseConfig = $config->getDatabaseConfig();
+        $this->connection = $this->createConnection();
+    }
+
+    public function setSession(Config $config): void
+    {
+        $sessionVariables = [];
+        $sessionVariables['QUERY_TAG'] = sprintf("'%s'", json_encode(['runId' => getenv('KBC_RUNID')]));
+
+        if ($config->getQueryTimeout()) {
+            $sessionVariables['STATEMENT_TIMEOUT_IN_SECONDS'] = $config->getQueryTimeout();
+        }
+
+        array_walk($sessionVariables, function (&$item, $key): void {
+            $item = vsprintf(
+                '%s=%s',
+                [
+                    $key,
+                    $item,
+                ]
+            );
+        });
+
+        $query = sprintf('ALTER SESSION SET %s;', implode(',', $sessionVariables));
+        $this->runRetriableQuery($query, 'alter session');
+    }
+
+    public function processSteps(array $steps): void
+    {
+        foreach ($steps as $step) {
+            if ($step['execution'] === 'parallel') {
```

Comment:
tohle je tady asi zatím zbytečný :-)

Reply:
jj zbytečný to je - udělal jsem to jen jako přípravu do budoucna :)

Reply:
je to pryč